src/workerd/api/streams/standard.c++

```diff
@@ -2996,19 +3043,28 @@ private:
         // case and handle appropriately (generally by canceling the readable
         // and exiting the loop).
         return read(js).then(js, ioContext.addFunctor(
-            [](auto& js, ReadResult result) mutable -> Result {
+            [&accumulator=*accumulator](auto& js, ReadResult result) mutable -> Result {
```

Comment:
Optional: this change from `kj::Own<Accumulator>` to `Accumulator&` looks more unsafe than it need to be.

If a future change released the `kj::Own<Accumulator>` between the call to read and the time that `then` here is invoked, then it would be obvious because the target is `nullptr` when the lambda executes. As-is, it would be a stale address dereference in the lambda which might crash or behave unexpectedly.

Reply:
Ignore this comment, overthinking it.